app/assets/javascripts/components/overview/available_actions.jsx

```diff
@@ -84,16 +88,24 @@ const AvailableActions = createReactClass({
   },
 
   enableChat() {
-    if (confirm('Are you sure you want to enable chat?')) {
-      return ChatActions.enableForCourse(this.state.course.id);
-    }
+    const course = this.state.course.id;
+    const onConfirm = function () {
+      return ChatActions.enableForCourse(course);
+    };
+    const confirmMessage = 'Are you sure you want to enable chat?';
+    this.props.initiateConfirm(confirmMessage, onConfirm);
   },
 
   enableRequests() {
-    if (confirm('Are you sure you want to enable the account requests?')) {
-      this.props.enableAccountRequests(this.state.course);
-      return alert(I18n.t('courses.accounts_generation_enabled'));
-    }
+    const enableRequests = this.props.enableAccountRequests;
+    const course = this.state.course;
+    const onConfirm = function () {
+      enableRequests(course);
+      alert(I18n.t('courses.accounts_generation_enabled'));
```

Comment:
Instead of an alert, this would be nicer as a Notification. Part of benefit of switching from 'confirm' to 'initiateConfirm' is that it avoids the slightly jarring UI of the browser-based popup dialogs.

Reply:
This is also where you could send another action to update the flag, or re-fetch the course from the API, so that the 'enable' button goes away.